src/storagesync/azext_storagesync/_params.py

```diff
@@ -0,0 +1,159 @@
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('storage_sync_service_name', storage_sync_service_name_type, options_list=['--name', '-n'])
+        c.argument('location', arg_type=get_location_type(self.cli_ctx),
+                   validator=get_default_location_from_resource_group)
+        c.argument('tags', tags_type, default={})  # tags must be at least an empty dictionary instead of None. (May be a bug in service and will check with service team.)
+
+    with self.argument_context('storagesync storage-sync-service delete') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('storage_sync_service_name', storage_sync_service_name_type, options_list=['--name', '-n'])
+
+    with self.argument_context('storagesync storage-sync-service show') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('storage_sync_service_name', storage_sync_service_name_type, options_list=['--name', '-n'])
+
+    with self.argument_context('storagesync storage-sync-service list') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+
+    with self.argument_context('storagesync sync-group create') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('storage_sync_service_name', storage_sync_service_name_type)
+        c.argument('sync_group_name', sync_group_name_type, options_list=['--name', '-n'])
+
+    with self.argument_context('storagesync sync-group delete') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('storage_sync_service_name', storage_sync_service_name_type)
+        c.argument('sync_group_name', sync_group_name_type, options_list=['--name', '-n'])
+
+    with self.argument_context('storagesync sync-group show') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('storage_sync_service_name', storage_sync_service_name_type)
+        c.argument('sync_group_name', sync_group_name_type, options_list=['--name', '-n'])
+
+    with self.argument_context('storagesync sync-group list') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('storage_sync_service_name', storage_sync_service_name_type)
+
+    with self.argument_context('storagesync cloud-endpoint create') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('storage_sync_service_name', storage_sync_service_name_type)
+        c.argument('sync_group_name', sync_group_name_type)
+        c.argument('cloud_endpoint_name', cloud_endpoint_name_type, options_list=['--name', '-n'])
+        c.argument('storage_account_resource_id', storage_account_type)
+        c.argument('azure_file_share_name', azure_file_share_name_type)
+        c.argument('storage_account_tenant_id', storage_account_tenant_id_type)
```

Comment:
can we get this tenant id by `storage_account_resource_id`?

Reply:
Yes, I do it in validator

---
In reply to: [392868993](https://github.com/Azure/azure-cli-extensions/pull/1389#discussion_r392868993) [](ancestors = 392868993)